ci/scripts/e2e-test-serial.sh

```diff
@@ -105,11 +108,9 @@ risedev slt -p 4566 -d dev './e2e_test/database/prepare.slt'
 risedev slt -p 4566 -d test './e2e_test/database/test.slt'
 
 echo "--- e2e, $mode, python_client"
-python3 -m pip install --break-system-packages psycopg
```

Comment:
This is unpinned, compared to the version in requirements.txt. Ends up installling a different version:
```
e2e, ci-3streaming-2serving-3fe, python_client
Collecting psycopg
  Downloading psycopg-3.2.5-py3-none-any.whl.metadata (4.4 kB)
```

And since we reuse the package cache, we may end up reusing this psycopg, whose binary hash will mismatch due to being a different version.